floris/tools/layout_functions.py

```diff
@@ -1,376 +1,323 @@
+    if wd is None:
+        wd = fi.floris.flow_field.wind_directions[0]
 
-def turbineDist(df, turbList):
-    """
-    Derive distance between any two turbines.
+    # Rotate yaw angles to inertial frame for plotting turbines relative to wind direction
+    yaw_angles = yaw_angles - wind_delta(np.array(wd))
 
-    Args:
-        df (pd.DataFrame): DataFrame with layout data.
-        turbList (list): list of 2 turbines for which spacing distance
-            is of interest.
+    if color is None:
+        color = "k"
 
-    Returns:
-        float: distance between turbines.
-    """
-    x1 = df.loc[turbList[0], "x"]
-    x2 = df.loc[turbList[1], "x"]
-    y1 = df.loc[turbList[0], "y"]
-    y2 = df.loc[turbList[1], "y"]
+    # If yaw angles is not 1D, assume we want first findex
+    yaw_angles = np.array(yaw_angles)
+    if yaw_angles.ndim == 2:
+        yaw_angles = yaw_angles[0, :]
 
-    dist = np.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)
+    rotor_diameters = fi.floris.farm.rotor_diameters.flatten()
+    for x, y, yaw, d in zip(fi.layout_x, fi.layout_y, yaw_angles, rotor_diameters):
+        R = d / 2.0
+        x_0 = x + np.sin(np.deg2rad(yaw)) * R
+        x_1 = x - np.sin(np.deg2rad(yaw)) * R
+        y_0 = y - np.cos(np.deg2rad(yaw)) * R
+        y_1 = y + np.cos(np.deg2rad(yaw)) * R
+        ax.plot([x_0, x_1], [y_0, y_1], color=color)
 
-    return dist
+    return ax
 
 
-def wakeAngle(df, turbList):
+def get_wake_direction(x_i: float, y_i: float, x_j: float, y_j: float) -> float:
     """
-    Get angles between turbines in wake direction
+    Calculates the wind direction at which the wake of turbine i would impact turbine j.
 
     Args:
-        df (pd.DataFrame): DataFrame with layout data.
-        turbList (list): list of 2 turbines for which spacing distance
-            is of interest.
+        x_i (float): X-coordinate of turbine i (the upstream turbine).
+        y_i (float): Y-coordinate of turbine i.
+        x_j (float): X-coordinate of turbine j (the downstream turbine).
+        y_j (float): Y-coordinate of turbine j.
 
     Returns:
-        wakeAngle (float): angle between turbines relative to compass
+        float: Wind direction in degrees (0-360) where 0 degrees represents wind
+               blowing from the north, and the angle increases clockwise.
     """
-    x1 = df.loc[turbList[0], "x"]
-    x2 = df.loc[turbList[1], "x"]
-    y1 = df.loc[turbList[0], "y"]
-    y2 = df.loc[turbList[1], "y"]
-    wakeAngle = (
-        np.arctan2(y2 - y1, x2 - x1) * 180.0 / np.pi
-    )  # Angle in normal cartesian coordinates
 
-    # Convert angle to compass angle
-    wakeAngle = 270.0 - wakeAngle
-    if wakeAngle < 0:
-        wakeAngle = wakeAngle + 360.0
-    if wakeAngle > 360:
-        wakeAngle = wakeAngle - 360.0
+    dx = x_j - x_i
+    dy = y_j - y_i
+
+    angle_rad = np.arctan2(dy, dx)
+    angle_deg = 270 - np.rad2deg(angle_rad)
+
+    # Adjust for "from" direction (add 180 degrees) and wrap within 0-360
```

Comment:
I'm not sure whether this comments matches the code below. Is this implicit in the %?

Reply:
Should've been on the line above, moving now